encode_test.go

```diff
@@ -75,8 +77,8 @@ func TestMarshal(t *testing.T) {
 		{"empty slice", []H{}, nil, false},
 		{"pointer", &H{"foo", 1}, []byte("foo  1    "), false},
 		{"nil", nil, nil, false},
-		{"invalid type", true, nil, true},
-		{"invalid type in struct", H{"foo", true}, nil, true},
+		{"invalid type", invtype, nil, true},
+		{"invalid type in struct", H{"foo", invtype}, nil, true},
```

Comment:
@ianlopshire not sure if this is 100% correct solution to check for incorrect type. Since bool now if correct type we need something else for this check

Reply:
This is a great solution!